### PR TITLE
Improve MP Config properties source and properties component

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: components/camel-properties/src/main/java/org/apache/camel/component/properties/PropertiesComponent.java [hunk 1]
```diff
@@ -184,7 +187,8 @@ public Properties loadProperties() {
         }
 
         if (!sources.isEmpty()) {
-            for (PropertiesSource ps : sources) {
+            for (int i = sources.size(); i-- > 0; ) {
```
Comment (davsclaus) [inline components/camel-properties/src/main/java/org/apache/camel/component/properties/PropertiesComponent.java]: Can you add a code commnet why we use revert ordering here

Comment (lburgazzoli) [reply components/camel-properties/src/main/java/org/apache/camel/component/properties/PropertiesComponent.java]: done

**Comment by lburgazzoli** [comment]
@davsclaus this PR should be complete now

File: components/camel-microprofile-config/src/main/java/org/apache/camel/component/microprofile/config/CamelMicroProfilePropertiesSource.java [hunk 2]
```diff
@@ -41,15 +42,51 @@ public String getProperty(String name) {
...
     @Override
-    protected void doStart() throws Exception {
+    public Properties loadProperties() {
+        if (config == null) {
+            config = ConfigProvider.getConfig();
+        }
+
+        Properties answer = new Properties();
+
+        for (String key: config.getPropertyNames()) {
```
Comment (davsclaus) [inline components/camel-microprofile-config/src/main/java/org/apache/camel/component/microprofile/config/CamelMicroProfilePropertiesSource.java]: Does this pass checkstyle?

Comment (lburgazzoli) [reply components/camel-microprofile-config/src/main/java/org/apache/camel/component/microprofile/config/CamelMicroProfilePropertiesSource.java]: it appears to do it

**Pull request merged**